controllers/providers/aws/autoscaling.go

```diff
@@ -54,17 +54,23 @@ func (a *AmazonClientSet) TerminateInstance(instance *autoscaling.Instance) erro
 	return nil
 }
 
-func (a *AmazonClientSet) SetInstanceStandBy(instance *autoscaling.Instance, scalingGroupName string) error {
-	instanceID := aws.StringValue(instance.InstanceId)
-	input := &autoscaling.EnterStandbyInput{
-		AutoScalingGroupName:           aws.String(scalingGroupName),
-		InstanceIds:                    aws.StringSlice([]string{instanceID}),
-		ShouldDecrementDesiredCapacity: aws.Bool(false),
+func (a *AmazonClientSet) SetInstanceStandBy(instances []*autoscaling.Instance, scalingGroupName string) (int, error) {
+	var instanceIDs []string
+	for _, instance := range instances {
+		if aws.StringValue(instance.LifecycleState) == autoscaling.LifecycleStateInService {
+			instanceIDs = append(instanceIDs, aws.StringValue(instance.InstanceId))
+		}
 	}
-
-	if _, err := a.AsgClient.EnterStandby(input); err != nil {
-		return err
+	if len(instanceIDs) > 0 {
+		input := &autoscaling.EnterStandbyInput{
+			AutoScalingGroupName:           aws.String(scalingGroupName),
+			InstanceIds:                    aws.StringSlice(instanceIDs),
+			ShouldDecrementDesiredCapacity: aws.Bool(false),
+		}
+		if _, err := a.AsgClient.EnterStandby(input); err != nil {
+			return len(instanceIDs), err
+		}
 	}
 
-	return nil
+	return len(instanceIDs), nil
```

Comment:
would a better pattern be returning (bool, error) ?
Do you actually need the number, or do you just need to know if instances were set to standby?

Reply:
This makes sense. I can use bool